daemon/cluster/services.go

```diff
@@ -262,6 +262,11 @@ func (c *Cluster) ServiceLogs(ctx context.Context, input string, config *backend
 		c.mu.RUnlock()
 		return err
 	}
+	// TODO(dperny) can I trust that service.Spec.Task always exists?
```

Comment:
You can - however you can't trust `service.Spec.Task.GetContainer()` to return a container.

You might get `nil` here. Please return a `Service logs only supported for containers` at that point